backend/apps/matching/admin.py

```diff
@@ -61,3 +61,46 @@ def logged_in_not_permitted(function=None, login_url="/"):
     if function:
         return actual_decorator(function)
     return actual_decorator
+
+
+def profile_visible_for_B(u):
+    return u.is_B
+
+
+def profile_visible_for_other_A(u):
+    return u.is_A
+
+
+def profile_visible_for_A(u):
+    return u.is_A
+
+
+def profile_visible_for_other_B(u):
+    return u.is_B
+
+
+def profile_own(uuid_accessed, uuid_own):
+    return uuid_accessed == uuid_own
+
+
+def required_at_least_one(restrictions_A, restrictions_B, function=None):
+    def actual_decorator(function):
+        def test(request, p_type, uuid, *args, **kwargs):
+            u = request.user
+            passes_test = False
+            if p_type == "A":
+                for restriction in restrictions_A:
+                    passes_test += restriction(u)
+            elif p_type == "B":
+                for restriction in restrictions_B:
+                    passes_test += restriction(u)
+            passes_test += profile_own(uuid_accessed=uuid, uuid_own=u.participant().info.uuid)
+            if passes_test > 0:
+                return function(request, p_type, uuid, *args, **kwargs)
+            raise Http404
```

Comment:
@bjrne Did you also implement something like this for 403 or what's the way to do it?

Reply:
Well a 403 page is implemented now, but you would need to link to it via its name "403". Http404 is the only Exception implemented by django in this way unfortunately:

https://github.com/django/django/blob/3d16496037fbb8a6bbc6b6e354fa4f5eb65e6cea/django/http/response.py#L539